lib/agent.js

```diff
@@ -749,3 +793,22 @@ Agent.prototype.registerMetric = function (name, labelsOrCallback, callback) {
 
   this._metrics.getOrCreateGauge(name, callback, labels)
 }
+
+/**
+ * Return true iff the given metric name is "disable", according to the
+ * `disableMetrics` config var.
+ *
+ * @returns {boolean}
+ */
+Agent.prototype._isMetricNameDisabled = function (name) {
+  if (!this._conf.disableMetricsRegExp) {
```

Comment:
This property is initialised as an empty array in `config.js` file. I think if we can ensure that the type stays the same (array) we can check on the length of it for early returns.

Reply:
There is this one example of a "mock" Agent._conf object that doesn't have `.disableMetricsRegExp` set:
https://github.com/elastic/apm-agent-nodejs/blob/trentm/otel-metrics-poc/test/lambda/mock/agent.js#L19-L23
Ideally that example would have a more complete default `Config` instance from "lib/config.js".

However, tests pass when I comment out the `if (!this._conf.disableMetricsRegExp) {` guard, so things are fine right now.

Reply:
I dropped the guard in commit 3feae2ba.